test/clients/services_manager_test.rb

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative '../test_helper'
+
+require "yast"
+require "services-manager/clients/services_manager"
+
+describe Y2ServicesManager::Clients::ServicesManager do
+  include Yast::UIShortcuts
+
+  # Finds in the widgets tree a widget with the given id
+  #
+  # @param tree [Yast::Term]
+  # @param id [Symbol]
+  #
+  # @return [Boolean]
+  def exist_widget?(tree, id)
+    return false unless tree.is_a?(Yast::Term)
+    return true if tree.value == :id && tree.params.first == id
+
+    tree.params.any? { |t| exist_widget?(t, id) }
```

Comment:
you can use nested find of Term https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/term.rb#L69
so something like 
```
return false unless tree.is_a?(Yast::Term)
!tree.nested_find { |t| t.value == :id && t[0] == id }.nil?
```